numpy_financial/_financial.py

```diff
@@ -302,6 +302,14 @@ def nper(rate, pmt, pv, fv=0, when='end'):
     return nper_array
 
 
+def _value_like(arr, value):
+    entry = arr.item(0)
+    if isinstance(entry, Decimal):
+        return Decimal(value)
+    else:
+        return value
```

Comment:
nit: should value here be case to the same dtype as entry?

Reply:
Good point; fixed.